src/subscription/subscribe.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2017, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+import {
+  addPath,
+  buildExecutionContext,
+  collectFields,
+  defaultFieldResolver,
+  execute,
+  getFieldDef,
+  getOperationRootType,
+} from '../execution/execute';
+import { getArgumentValues } from '../execution/values';
+import { GraphQLSchema } from '../type/schema';
+import invariant from '../jsutils/invariant';
+import mapAsyncIterator from './mapAsyncIterator';
+
+import type {
+  ExecutionContext,
+  ExecutionResult,
+} from '../execution/execute';
+import type {
+  DocumentNode,
+  OperationDefinitionNode,
+} from '../language/ast';
+import type { GraphQLResolveInfo } from '../type/definition';
+
+
+/**
+ * Implements the "Subscribing to request" section of the GraphQL specification.
+ *
+ * Returns an AsyncIterator
+ *
+ * If the arguments to this function do not result in a legal execution context,
+ * a GraphQLError will be thrown immediately explaining the invalid input.
+ */
+export function subscribe(
+  schema: GraphQLSchema,
+  document: DocumentNode,
+  rootValue?: mixed,
```

Comment:
Same purpose as in `execute` - to provide the top level value context for performing the operation.

In practice a common use of this is to store root field resolvers, and the test case leverages this ability by placing the resolver function for `importantEmail` on the rootValue